crates/fluvio-version-manager/src/common/settings.rs

```diff
@@ -0,0 +1,66 @@
+use std::fs::write;
+use std::path::PathBuf;
+
+use color_eyre::eyre::{Error, Result};
+use serde::{Deserialize, Serialize};
+use semver::Version;
+
+use fluvio_hub_util::fvm::Channel;
+
+use super::workdir::fvm_workdir_path;
+
+pub const SETTINGS_TOML_FILENAME: &str = "settings.toml";
+
+/// The `settings.toml` is in charge of keeping track of the active version
+/// through the default key, which holds the name of the directory under
+/// `~/.fvm/pkgset/default/versions` for the desired default version.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Settings {
+    /// The active `channel` for the Fluvio Installation
+    pub channel: Option<Channel>,
```

Comment:
this should be enum? Since it'e either channel or version

Reply:
Yeah, it is `enum`. Channel is an enum with either `Stable`, `Latest` or `Version`.

Here is the source for when we introduced it in the previous PR: https://github.com/infinyon/fluvio/commit/f72af5d9c35fa91359c013445651f76a98fe9206#diff-c1995b0a457b64145b5f1912f0b961dfc7116178a89908cb2580ba3ad903f142R35-R42

Reply:
Then why it is repeated here?

Reply:
Is imported from Hub's FVM

Reply:
Let's have a single definition

Reply:
this needs to be fixed.  it is probably most critical part of this PR

Reply:
But the definition its being reused no? The one in Hub Util crate.

Or are you referring to some other?

Reply:
As a mater of fact, the `Channel` struct present here is the one used by FVM to determine 3 possible states:

- Latest
- Stable
- A specific SemVer Compliant version number

This is the same enum used for FVM API Client.

The implementation lives in the Hub Util crate and is being imported to be used here, given that is also used in the Settings toml

Reply:
I am still trying to understand why there are two different semantics difference

Reply:
This is a struct while the other is an enum.   And it seems to have duplicate field `version` associated with it.

Reply:
it would be good to have tests to show what it does